### PR TITLE
add confilct_resolver_max_attempts property

### PR DESCRIPTION
**Pull request by JonasFrey96**
PR for Issue #240 
Added a simple property to the ConflictResolver that can be changed if needed.

### CODE REVIEW

File: simple_parsing/conflicts.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ def unflatten(possibly_related_wrappers: list[DataclassWrapper]) -> list[Datacla
 class ConflictResolver:
     def __init__(self, conflict_resolution=ConflictResolution.AUTO):
         self.conflict_resolution = conflict_resolution
+        self._max_attempts = 50
```
Comment (lebrice) [inline simple_parsing/conflicts.py:64]:
```suggestion
        self.max_attempts = 50
```

File: simple_parsing/conflicts.py [hunk 2]
```diff
@@ -385,3 +386,11 @@ def _conflict_exists(self, all_wrappers: list[DataclassWrapper]) -> bool:
                         return True
                     arg_names.add(option)
         return False
+
+    @property
+    def max_attempts(self) -> int:
+        return self._max_attempts
+    
+    @max_attempts.setter
+    def max_attempts(self, value: int):
```
Comment (lebrice) [inline simple_parsing/conflicts.py:395]: No need for a property here:
```suggestion
```

File: simple_parsing/conflicts.py [hunk 3]
```diff
@@ -106,9 +107,9 @@ def resolve_and_flatten(self, wrappers: list[DataclassWrapper]) -> list[Dataclas
 
             conflict = self.get_conflict(wrappers_flat)
             cur_attempts += 1
-            if cur_attempts == max_attempts:
+            if cur_attempts == self._max_attempts:
                 raise ConflictResolutionError(
-                    f"Reached maximum number of attempts ({max_attempts}) "
+                    f"Reached maximum number of attempts ({self._max_attempts}) "
```
Comment (lebrice) [inline simple_parsing/conflicts.py:112]:
```suggestion
                    f"Reached maximum number of attempts ({self.max_attempts}) "
```

File: simple_parsing/conflicts.py [hunk 4]
```diff
@@ -106,9 +107,9 @@ def resolve_and_flatten(self, wrappers: list[DataclassWrapper]) -> list[Dataclas
 
             conflict = self.get_conflict(wrappers_flat)
             cur_attempts += 1
-            if cur_attempts == max_attempts:
+            if cur_attempts == self._max_attempts:
```
Comment (lebrice) [inline simple_parsing/conflicts.py:110]:
```suggestion
            if cur_attempts == self.max_attempts:
```

File: simple_parsing/conflicts.py [hunk 5]
```diff
@@ -385,3 +386,5 @@ def _conflict_exists(self, all_wrappers: list[DataclassWrapper]) -> bool:
                         return True
                     arg_names.add(option)
         return False
+
+        self._max_attempts = value
```
Comment (lebrice) [inline simple_parsing/conflicts.py:390]:
```suggestion
```

File: simple_parsing/conflicts.py [hunk 6]
```diff
@@ -384,4 +385,4 @@ def _conflict_exists(self, all_wrappers: list[DataclassWrapper]) -> bool:
                     if option in arg_names:
                         return True
                     arg_names.add(option)
-        return False
+        return False
```
Comment (lebrice) [inline simple_parsing/conflicts.py:388]:
```suggestion
        return False

```

**Comment by JonasFrey96** [comment]
This looks great. Thanks a lot. 
Can you merge the PR using the CLI given that from my understanding I cannot verify the commit in hindsight ?

**Pull request merged**